flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/table/PulsarTableOptions.java

```diff
@@ -0,0 +1,282 @@
+                                            code("sink.custom-topic-router"))
+                                    .build());
+
+    public static final ConfigOption<Duration> SINK_MESSAGE_DELAY_INTERVAL =
+            ConfigOptions.key("sink.message-delay-interval")
+                    .durationType()
+                    .defaultValue(Duration.ZERO)
+                    .withDescription(
+                            "(Optional) the message delay delivery interval that is used in the [Pulsar DataStream sink connector](https://nightlies.apache.org/flink/flink-docs-master/docs/connectors/datastream/pulsar/#pulsar-sink).");
+
+    // --------------------------------------------------------------------------------------------
+    // Format Options
+    // --------------------------------------------------------------------------------------------
+
+    public static final ConfigOption<String> KEY_FORMAT =
+            ConfigOptions.key("key" + FORMAT_SUFFIX)
+                    .stringType()
+                    .noDefaultValue()
+                    .withDescription(
+                            "Defines the format identifier for decoding/encoding key bytes in "
+                                    + "Pulsar message. The identifier is used to discover a suitable format factory.");
+
+    public static final ConfigOption<List<String>> KEY_FIELDS =
+            ConfigOptions.key("key.fields")
+                    .stringType()
+                    .asList()
+                    .defaultValues()
+                    .withDescription(
+                            "An explicit list of physical columns from the table schema that are decoded/encoded from the key bytes of a Pulsar message. By default, this list is empty and thus a key is undefined.");
+
+    public static final ConfigOption<String> VALUE_FORMAT =
+            ConfigOptions.key("value" + FORMAT_SUFFIX)
+                    .stringType()
+                    .noDefaultValue()
+                    .withDescription(
+                            "Defines the format identifier for decoding/encoding value data. "
+                                    + "The identifier is used to discover a suitable format factory.");
+
+    // --------------------------------------------------------------------------------------------
+    // Pulsar Options
+    // --------------------------------------------------------------------------------------------
+
+    /**
+     * Exactly same as {@link
+     * org.apache.flink.connector.pulsar.common.config.PulsarOptions#PULSAR_ADMIN_URL}. Copied here
+     * because it is a required config option and should not be included in the {@link
+     * org.apache.flink.table.factories.FactoryUtil.FactoryHelper#validateExcept(String...)} method.
+     *
+     * <p>By default all {@link org.apache.flink.connector.pulsar.common.config.PulsarOptions} are
+     * included in the validateExcept() method./p>
+     */
+    public static final ConfigOption<String> ADMIN_URL =
```

Comment:
we should look into removing the admin url

Reply:
Yeah. I saw your issue here - https://issues.apache.org/jira/browse/FLINK-32938.

This is an inherited issue that we should handle it separately.